src/caesium/byte_bufs.clj

```diff
@@ -39,11 +39,13 @@
 (defprotocol BufLen
   (^Long buflen [this]))
 
-(extend-protocol BufLen
-  (Class/forName "[B")
-  (buflen [this]
-    (long (alength ^bytes this)))
+(extend (Class/forName "[B")
+  BufLen
+  {:buflen
+   (fn [^bytes this]
+     (long (alength this)))})
```

Comment:
I don't really understand this change: what was wrong with the previous code?

Reply:
It triggered a `wrong-tag` warning with eastwood and puzzled me initially as well. It turns out the `extend-protocol` macro is expanded in such a way that the `(Class/forName "[B")` part becomes a type hint tag, and such a tag is invalid. Here's a relevant bit from the eastwood docs:

> However, if you try to use extend-type or extend-protocol with an expression that evaluates to a class at run time, e.g. (Class/forName "[D") as the type, most things will work correctly, but it will also expand to code that has that expression as a type tag on the first argument of all functions. That expression is not a valid type tag. Clojure silently ignores such type tags, so there are no errors or warnings during compilation. Since the invalid type tag is ignored, you will be disappointed if you were relying on it to avoid reflection.

Source: https://github.com/jonase/eastwood#wrong-tag-warnings-in-uses-of-extend-type-and-extend-protocol-macro